### PR TITLE
feature/spritekit

### PR DESCRIPTION
**Pull request by robha141**

### CODE REVIEW

File: Spritekit/Final/FlappyBird/Game/Nodes/PipeNode.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+//
+
+import SpriteKit
+
+final class PipeNode: SKNode {
+    // MARK: Properties
+    private let holeHeight: CGFloat
+    
+    let width: CGFloat
+    /// Value, which says if the bird already flew over the pipe.
```
Comment (robha141) [inline Spritekit/Final/FlappyBird/Game/Nodes/PipeNode.swift]:
```suggestion
    /// Value, which says if the bird already flew through the pipe.
```

File: Spritekit/Final/FlappyBird/Game/GameStates/GameStateInitial.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+//  FlappyBird
+//
+//  Created by Róbert Oravec on 14.11.2023.
+//
+
+import GameplayKit
+
+final class GameStateInitial: GKState {
+    // MARK: Properties
+    private unowned var gameViewController: GameViewController
```
Comment (martinvidovic) [inline Spritekit/Final/FlappyBird/Game/GameStates/GameStateInitial.swift]: why do we need unowned when inheriting from `GFState`?

File: Spritekit/Final/FlappyBird/Game/Nodes/PipeNode.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+import SpriteKit
+
+final class PipeNode: SKNode {
+    // MARK: Properties
+    private let holeHeight: CGFloat
+    
+    let width: CGFloat
+    /// Value, which says if the bird already flew through the pipe.
+    var scoreCounted = false
```
Comment (martinvidovic) [inline Spritekit/Final/FlappyBird/Game/Nodes/PipeNode.swift]: probably I will find it elsewhere, but pls check if they can be private

**Review by martinvidovic — APPROVED** [review]
just a small comments/suggestions, otherwise good ❤️

**Pull request merged**